CGAL header-only errors

96 errors in the following testsuite, all of the form "undefined reference to <something in Boost>":

```
    354 undefined reference to `boost::system::system_category()'
    392 undefined reference to `boost::system::generic_category()'
   1443 undefined reference to `boost::detail::get_tss_data(void const*)'
   2886 undefined reference to `boost::detail::set_tss_data(void const*, boost::shared_ptr<boost::detail::tss_cleanup_function>, void*, bool)'
```

See for example https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-I-150/AABB_tree_Demo/TestReport_lrineau_Fedora-strict-ansi.gz

It is probably related to the feature of CGAL that allows not to use Boost.Thread with C++11.

This bug blocks CGAL-4.9.

--- a/Installation/cmake/modules/CGAL_SetupBoost.cmake
+++ b/Installation/cmake/modules/CGAL_SetupBoost.cmake
@@ -13,7 +13,8 @@ if ( NOT CGAL_Boost_Setup )
           AND CMAKE_CXX_FLAGS MATCHES "\\-std=(c|gnu)\\+\\+[01][14yxz]")
         #GCC 6.0+ without c++03 on
         OR ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0
-             AND NOT CMAKE_CXX_FLAGS MATCHES "\\-std=(c|gnu)\\+\\+[90][83]")
+             AND NOT CMAKE_CXX_FLAGS MATCHES "\\-std=(c|gnu)\\+\\+[90][83]"
+             AND NOT CMAKE_CXX_FLAGS MATCHES "\\-ansi")
       ) )
     set ( CGAL_requires_Boost_libs FALSE )
   endif()

--- a/Installation/cmake/modules/CGAL_SetupBoost.cmake
+++ b/Installation/cmake/modules/CGAL_SetupBoost.cmake
@@ -6,10 +6,15 @@ if ( NOT CGAL_Boost_Setup )
   if ( DEFINED  MSVC_VERSION AND "${MSVC_VERSION}" GREATER 1800)
     set ( CGAL_requires_Boost_libs FALSE )
   endif()
+
   if ( CMAKE_COMPILER_IS_GNUCXX
       AND(
+        #CMAKE_CXX_STANDARD_DEFAULT is available starting cmake 3.4
+        ( DEFINED CMAKE_CXX_STANDARD_DEFAULT
+          AND ${CMAKE_CXX_STANDARD_DEFAULT} GREATER 03
+          AND ${CMAKE_CXX_STANDARD_DEFAULT} LESS 98 )
         #GCC 4.8+ with c++11 on
-        ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8
+        OR ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 4.8
           AND CMAKE_CXX_FLAGS MATCHES "\\-std=(c|gnu)\\+\\+[01][14yxz]")
         #GCC 6.0+ without c++03 on
         OR ( NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0

